hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c

```diff
@@ -177,9 +249,10 @@ nrf52_pwm_close(struct os_dev *odev)
 
     dev = (struct pwm_dev *) odev;
     inst_id = dev->pwm_instance_id;
+    assert(instances[inst_id].in_use);
 
     nrfx_pwm_uninit(&instances[inst_id].drv_instance);
-    instances[inst_id].playing = false;
+    cleanup_instance(inst_id);
```

Comment:
I take it the assert is whats currently protecting us from a regressions from this recent commit?
https://github.com/apache/mynewt-core/pull/821/files

Reply:
The problem which the commit you mentioned fixes is the misuse of in_use flag. When you reported the problem, what was "breaking" execution was in fact an assert on pwm_open. Only yesterday I noticed that fix is not coherent with how the flag should be used.